ui/app/AppLayouts/Wallet/panels/BuyCryptoProvidersListPanel.qml

```diff
@@ -0,0 +1,76 @@
+import QtQuick 2.14
+import QtQuick.Layouts 1.0
+import QtQml.Models 2.14
+import SortFilterProxyModel 0.2
+
+import StatusQ.Controls 0.1
+import StatusQ.Core 0.1
+
+import AppLayouts.Wallet.controls 1.0
+
+ColumnLayout {
+    id: root
+
+    // required properties
+    required property bool providersLoading
+    required property var providersModel
+    required property bool isUrlBeingFetched
+
+    // exposed api
+    property alias currentTabIndex: tabBar.currentIndex
+    signal providerSelected(var model)
+
+    QtObject {
+        id: d
+        readonly property int loadingItemsCount: 5
+    }
+
+    spacing: 20
+
+    StatusSwitchTabBar {
+        id: tabBar
+        objectName: "tabBar"
+        Layout.alignment: Qt.AlignHCenter
+        Layout.fillWidth: true
+        StatusSwitchTabButton {
+            text: qsTr("One time")
+        }
+        StatusSwitchTabButton {
+            text: qsTr("Recurrent")
+        }
+    }
+
+    StatusListView {
+        objectName: "providersList"
+        Layout.fillWidth: true
+        Layout.fillHeight: true
+
+        DelegateModel {
+            id: regularModel
+            model: SortFilterProxyModel {
+                sourceModel: root.providersModel
+                filters: ValueFilter {
+                    enabled: tabBar.currentIndex
+                    roleName: "supportsRecurrentPurchase"
+                    value: true
+                }
+            }
+            delegate: BuyCryptoProvidersDelegate {
+                isUrlLoading: root.isUrlBeingFetched
```

Comment:
hmm the URL being loaded for one provider makes all go into loading mode, that doesn't look very good. If it's not too hard, it would be clearer user feedback to only put the loading animation on the one clicked.

Reply:
aaah crap, yes for sure, I'll update this